### PR TITLE
refactored IsNumeric and IsInt

### PR DESCRIPTION
**Pull request by mgroves**
@AlexArchive here is one way that #45 could be done.

I've refactored IsNumeric to rely on the .NET framework instead of string parsing. I've made IsInt separate, but also relying on the .NET framework. One drawback is that very large integers can't be handled by IsInt.

Hope you find this helpful, thanks!

### CODE REVIEW

File: Validator.UnitTest/ValidatorTest.cs [hunk 1]
```diff
@@ -156,10 +161,15 @@ public void IsBase64(string input, bool expected)
...
         [Theory]
         [InlineData("123", true)]
-        [InlineData("50000000000000000000000000", true)]
+        [InlineData("9223372036854775807", true)] // long.MaxValue
+        [InlineData("9223372036854775808", false)] // long.MaxValue + 1
+        [InlineData("-9223372036854775808", true)] // long.MinValue
+        [InlineData("-9223372036854775809", false)] // long.MinValue - 1
         [InlineData("123.123", false)]
         [InlineData("", false)]
         [InlineData("", false)]
```
Comment (ledbutter) [inline Validator.UnitTest/ValidatorTest.cs]: @mgroves While you are at it, can you remove this duplicate test case?

File: Validator/IsInt.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        /// Determine whether input represents an integer
+        /// Can only check numbers as large as long.MaxValue or as small as long.MinValue
+        /// Commas are not valid
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static bool IsInt(string input)
+        {
+            long dummy;
+            return long.TryParse(input, out dummy);
```
Comment (ledbutter) [inline Validator/IsInt.cs]: @mgroves "Int" in `IsInt()` has special meaning in C#, I'd have expected this to be `int.TryParse()`. Are you trying to reconcile the difference with JS by calling it `IsInt()` but having it really mean--in a C# context--`IsLong()`?

**Comment by ledbutter** [comment]
@mgroves I really like how much cleaner this is now, let's talk about my comments and move forward.

**Comment by mgroves** [comment]
I was trying to partially preserve some meaning when I decided to use long. I believe the IsInt in validator.js will return true for *any* integer, no matter how large. That would take some string parsing or bringing in another library, so I just decided that long was probably good enough.

But, you bring up a good point that "int" has meaning in C#, so perhaps this method should become two methods: IsInt and IsLong

**Comment by ledbutter** [comment]
Closing as it has been open for over a year without any further updates

**Pull request closed without merging**